openml-h2o/src/main/java/com/feedzai/openml/h2o/params/ParametersBuilderUtil.java

```diff
@@ -128,9 +135,75 @@ private ParametersBuilderUtil() { }
+                // We only want the fields that are present in the list of fields of the algorithm class.
+                .filter(modelParam -> usefulFields.contains(modelParam.getName()))
                 .collect(Collectors.toSet());
     }
 
+    /**
+     * Returns the fields that are set in the given {@linkplain ModelParametersSchemaV3 algorithmClass} as useful for
+     * the algorithm.
+     *
+     * @param algorithmClass The algorithm class to get the useful fields.
+     * @param allParameters  All the parameters to be added if useful fields is not provided by the class.
+     * @return The {@link List} with the useful fields for the given algorithm class.
+     * @since @@@feedzai.next.release@@@
+     */
+    private static List<String> getUsefulFields(final Class<? extends ModelParametersSchemaV3> algorithmClass,
+                                                final Set<String> allParameters) {
+
+        final Field[] declaredFields = algorithmClass.getDeclaredFields();
+        final List<Field> staticFields = new ArrayList<>();
+
+        for (final Field field : declaredFields) {
+            if (isStatic(field.getModifiers())) {
+                staticFields.add(field);
+            }
+        }
+
+        final Optional<Field> usefulFieldsArray = staticFields.stream()
+                .filter(fields -> "fields".equalsIgnoreCase(fields.getName()))
+                .findFirst();
+
+        final ImmutableList.Builder<String> usefulFields = ImmutableList.builder();
+
+        if (usefulFieldsArray.isPresent()) {
+            usefulFields.add(getFieldContent(algorithmClass, usefulFieldsArray.get()));
+        } else {
+            usefulFields.addAll(allParameters);
+        }
+
+        return usefulFields.build();
+    }
+
+    /**
+     * Returns an array with content of the given {@link Field} for the given {@linkplain ModelParametersSchemaV3
+     * algorithmClass}.
+     *
+     * <p> If the given field is not a {@link String} array an empty array is returned.
+     *
+     * @param algorithmClass The class with the given {@code field}.
+     * @param field          The field to retrieve the value.
+     * @return The array with the field content.
+     * @since @@@feedzai.next.release@@@
+     */
+    private static String[] getFieldContent(final Class<? extends ModelParametersSchemaV3> algorithmClass,
+                                            final Field field) {
+
+        final Class<?> fieldType = field.getType();
+
+        if (fieldType.isArray() && fieldType.getComponentType().equals(String.class)) {
+            try {
+                return (String[]) field.get(null);
+            } catch (final IllegalAccessException e) {
+                throw new IllegalArgumentException(String.format(
+                        "Unable to get useful fields for model of type: %s", algorithmClass.getTypeName()
+                ));
+            }
+        }
+        return new String[0];
```

Comment:
Shouldn't we throw the same exception in the "else" part of the check above?
I'm not comfortable with this silent failure.

Reply:
This is not exactly a silent failure. I can log here that the `fields` field is not a String array, but we shouldn't throw an exception if for some reason an implementation of `ModelParametersSchemaV3` has a static field called `fields.
If you think we should ensure that if the fields exists it should have the content we expect it to have, then it makes sense to thrown an exception.

Reply:
I agree with logging in case something changes in the future.